src/requests/cache/mod.rs

```diff
@@ -0,0 +1,9 @@
+use crate::{MomentoResult};
+use crate::cache_client::CacheClient;
+
+pub mod set_add_elements;
+
+pub trait MomentoResponse {}
+pub trait MomentoRequest<R: MomentoResponse> {
+    fn send(self: Self, cache_client: &CacheClient) -> impl std::future::Future<Output = MomentoResult<R>> + Send;
+}
```

Comment:
`send()` is not the intended way of sending requests with this api right? To help make the api have 1 shape to follow did you consider something like

```suggestion
pub trait MomentoResponse {}
pub trait MomentoRequest {
    type Response: MomentoResponse;
}

trait SendableRequest<R: MomentoRequest> {
  fn send(self: Self, cache_client: &CacheClient) -> impl std::future::Future<Output = R::Response> + Send;
}
```

which lets you publicly advertise request and response type bindings without exposing the send() function to users?

Also, what purpose does `MomentoResponse` serve? Any reason not to just have
```rust
pub trait MomentoRequest {
    type Response;
}
```
?

Addressing this would simplify send_request to
```rust
pub async fn send_request<R: MomentoRequest>(...) -> MomentoResult<R::Response>
```
which is nice because (1) it only has 1 variadic and (2) there is only 1 valid variadic value of response per request kind anyway, and this expresses that constraint succinctly.

Reply:
this is the first i've heard/seen of `type`. I'll have to read up on that.

Reply:
I did not know about `type` - now that I've read up on it / played with it, I like that much better and love that we get rid of a level of generics. I also agree that there's probably no value in `MomentoResponse` now. Those changes are made.

w/rt separating the `send` out into a different trait, in theory I like that idea; in practice, in order for `CacheClient` to (a) have a public `send_request` fn, and (b) that `fn` to require a `SendableRequest` as a trait bound on its argument, it forces the `SendableRequest` to be public anyway (or else the compiler gives a warning), so I'm not 100% sure whether it's worth separating these traits. I agree that `MomentoRequest.send` isn't the expected public API, but I don't necessarily care if curious users end up calling it, so not 100% sure if it is worth splitting out into 2 traits. But I did it in https://github.com/momentohq/client-sdk-rust/pull/175/commits/fb50255179e50212ff6ade6de83f0ed55f0da307 anyway just to see what it looks like.